packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -1,273 +1,70 @@
+} from '../../_modules_/grid';
+import { GridContextProvider } from '../../_modules_/grid/context/GridContextProvider';
+import { useDataGridComponent } from './useDataGridComponent';
+import { DataGridProps, MAX_PAGE_SIZE } from './DataGridProps';
 
-const FORCED_PROPS: Partial<GridComponentProps> = {
+const DATA_GRID_FORCED_PROPS: Omit<
+  GridComponentProps,
+  Exclude<keyof DataGridProps, 'pagination'>
+> = {
+  apiRef: undefined,
   disableColumnResize: true,
   disableColumnReorder: true,
   disableMultipleColumnsFiltering: true,
   disableMultipleColumnsSorting: true,
   disableMultipleSelection: true,
   pagination: true,
-  apiRef: undefined,
   onRowsScrollEnd: undefined,
   checkboxSelectionVisibleOnly: false,
 };
 
-export type DataGridProps = Omit<
-  GridComponentProps,
-  | 'apiRef'
-  | 'checkboxSelectionVisibleOnly'
-  | 'disableColumnResize'
-  | 'disableColumnReorder'
-  | 'disableMultipleColumnsFiltering'
-  | 'disableMultipleColumnsSorting'
-  | 'disableMultipleSelection'
-  | 'licenseStatus'
-  | 'options'
-  | 'onRowsScrollEnd'
-  | 'pagination'
-  | 'scrollEndThreshold'
-  | 'selectionModel'
-> & {
-  apiRef?: undefined;
-  checkboxSelectionVisibleOnly?: false;
-  disableColumnResize?: true;
-  disableColumnReorder?: true;
-  disableMultipleColumnsFiltering?: true;
-  disableMultipleColumnsSorting?: true;
-  disableMultipleSelection?: true;
-  onRowsScrollEnd?: undefined;
-  selectionModel?: GridRowId | GridRowId[];
-  pagination?: true;
-};
-
-const MAX_PAGE_SIZE = 100;
-
 const DataGridRaw = React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(
   inProps,
   ref,
 ) {
-  const props = useThemeProps({ props: inProps, name: 'MuiDataGrid' });
-  const { pageSize: pageSizeProp, selectionModel: dataGridSelectionModel, ...other } = props;
-
-  let pageSize = pageSizeProp;
-  if (pageSize && pageSize > MAX_PAGE_SIZE) {
-    pageSize = MAX_PAGE_SIZE;
+  if (inProps.pageSize! > MAX_PAGE_SIZE) {
+    throw new Error(`'props.pageSize' cannot exceed 100 in DataGrid.`);
   }
 
-  const selectionModel =
-    dataGridSelectionModel !== undefined && !Array.isArray(dataGridSelectionModel)
-      ? [dataGridSelectionModel]
-      : dataGridSelectionModel;
+  const props = React.useMemo<GridComponentProps>(
+    () => ({
+      ...inProps,
+      ...DATA_GRID_FORCED_PROPS,
+    }),
+    [inProps],
+  );
+
+  const apiRef = useGridApiRef();
+
+  useDataGridComponent(apiRef, props);
 
   return (
-    <GridComponent
-      ref={ref}
-      pageSize={pageSize}
-      {...other}
-      {...FORCED_PROPS}
-      selectionModel={selectionModel}
-      licenseStatus="Valid"
-    />
+    <GridContextProvider apiRef={apiRef} props={props}>
+      <GridRoot ref={ref}>
+        <GridErrorHandler>
+          <GridHeaderPlaceholder />
+          <GridBody />
+          <GridFooterPlaceholder />
+        </GridErrorHandler>
+      </GridRoot>
+    </GridContextProvider>
   );
 });
 
+DataGridRaw.defaultProps = DEFAULT_GRID_OPTIONS;
```

Comment:
If we re-add the `useThemeProps`, by setting the default props we won't get the theme translations yet. This default options object has a `localeText` prop with the English translation and the [`getThemeProps`](https://github.com/mui-org/material-ui/blob/bb3b9adbd42d8d0aa5f50b5da60f7ee377118a48/packages/material-ui-styles/src/getThemeProps/getThemeProps.js#L15-L17)  used internally doesn't overwrite the value if there's a default one.

Reply:
Good idea to test it correctly :+1:

Reply:
Fixed